CONTRIBUTING.md

```diff
@@ -98,11 +98,27 @@ If a flaky test cannot be fixed it should be removed from the suite.
 
 ## Testing new applications
 
-To test new applications you have to follow a similar process to
-[adding new tests](#adding-new-tests) however you will also need to configure
-this application to run the application through [docker compose][]
+To test new applications you will need to follow the
+[process for adding containers](docs/docker.md#adding-containers)
+to the [docker compose][].
 
-A brief guide to setting this up is available in
-[docs/docker.md](docs/docker.md#adding-containers).
+Once these are running inside of docker you should follow the
+[adding new tests](#adding-new-tests) process.
+
+When setting up the new RSpec tag such as `new_app: true` you will need to add
+a corresponding make `test-new-app` step that runs only tests with that tag.
+
+To run against specific revisions of the applications (such as commits and branches),
+you'll need to add an entry for the application being tested to the apps array in
+the Jenkinsfile.
+
+Once you have merged your tests into this repository and removed the new tag,
+because you are confident in them, you'll want to enable the tests to be run on
+every commit to the applications repositories.  The Jenkinsfile for most applications
+uses the `buildProject` which  has a `publishingE2ETests` parameter that enables this
+functionality.  Here is an [example of enabling][calendars-jenkinsfile] the E2E
+tests on calendars, and using the `PUBLISHING_E2E_TESTS_COMMAND` variable to
```

Comment:
Might be slightly easier to follow if the example is publisher since you kind of need to have domain knowledge to know that calendars is a an app that renders publisher content - without that it's a bit of wtf